src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -511,13 +529,23 @@ static void rr_connectivity_changed_locked(grpc_exec_ctx *exec_ctx, void *arg,
                                            grpc_error *error) {
   subchannel_data *sd = arg;
   round_robin_lb_policy *p = sd->subchannel_list->policy;
+  if (GRPC_TRACER_ON(grpc_lb_round_robin_trace)) {
+    gpr_log(
+        GPR_DEBUG,
+        "[RR %p] connectivity changed for subchannel %p, subchannel_list %p: "
+        "prev_state=%d new_state=%d p->shutdown=%d "
+        "sd->subchannel_list->shutting_down=%d error=%s",
+        (void *)p, (void *)sd->subchannel_list, (void *)sd->subchannel,
+        sd->prev_connectivity_state, sd->curr_connectivity_state, p->shutdown,
```

Comment:
I think you want to use `sd->pending_connectivity_state_unsafe` instead of `sd->curr_connectivity_state`, since the latter won't be updated until line 566 below.